modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/QueryContext.java

```diff
@@ -143,13 +155,36 @@ public IndexingQueryFilter filter() {
         return memTracker;
     }
 
+    /** {@inheritDoc} */
+    @Override public GroupByData newGroupByDataInstance(Session ses, ArrayList<Expression> expressions,
+        boolean isGrpQry, int[] grpIdx) {
+        if (memTracker == null)
+            return null;
+
+        boolean spillingEnabled = Boolean.getBoolean(IgniteSystemProperties.IGNITE_SQL_USE_DISK_OFFLOAD);
```

Comment:
This is synchronous call to HashTable.
Let's make this value static.

Reply:
If we run multiple tests with offload enabled/disabled, this constant would be initialized only once, while we need it has different values for different tests, so this is the reason why it is not static.

Reply:
Is it possible to cache value in some MemoryManager field and pass copy actual to context?
This will allow to change value between node starts and avoid unnecessary synchronous calls.